src/modules/navigator/mission_block.cpp

```diff
@@ -100,7 +100,13 @@ MissionBlock::is_mission_item_reached_or_completed()
 	// Indefinite Waypoints
 	case NAV_CMD_LAND: /* fall through */
 	case NAV_CMD_VTOL_LAND:
-		return _navigator->get_land_detected()->landed;
+		if (_navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING) {
+			// handled below
+			break;
```

Comment:
Are you sure this works? Now in FW the waypoint gets accepted when only horizontally at the landing point. This might be good for VTOL but for FW? It might accept it now before it actually is on the ground, no? Maybe only do this for VTOL land and not for land?

Reply:
@KonradRudin I removed it while doing my investigation for the bug. Did not fix the bug but still will be gone.